File: vibebot/workers/digest_builder.py
import logging

log = logging.getLogger(__name__)


def run(plugins: list, conn, llm) -> None:
    """Stage 2: Call build_digest() on each enabled plugin."""
    for plugin in plugins:
        msg_id = plugin.build_digest(conn, llm)
        if msg_id is not None:
            log.info(
                "Plugin '%s' built digest → outbound_message id=%d.",
                plugin.name,
                msg_id,
            )
        else:
            log.info("Plugin '%s': no new items to digest.", plugin.name)

File: src/workers/digestBuilder.ts
import Database from 'better-sqlite3';
import { BaseChatModel } from '@langchain/core/language_models/chat_models';
import { BasePlugin } from '../core/basePlugin';

export async function run(
  plugins: BasePlugin[],
  db: Database.Database,
  llm: BaseChatModel,
): Promise<void> {
  for (const plugin of plugins) {
    const msgId = await plugin.buildDigest(db, llm);
    if (msgId !== null) {
      console.info(`Plugin '${plugin.name}' built digest → outbound_message id=${msgId}.`);
    } else {
      console.info(`Plugin '${plugin.name}': no new items to digest.`);
    }
  }
}
